MobileWallet/Common/NotificationManager.swift

```diff
@@ -54,22 +85,51 @@ class NotificationManager {
         }
 
         notificationCenter.requestAuthorization(options: options) {
-            (didAllow, error) in
+            (_, error) in
             guard error == nil else {
                 TariLogger.error("NotificationManager request authorization", error: error)
+
                 completionHandler(false)
                 return
             }
 
-            completionHandler(didAllow)
+            self.registerWithAPNS(completionHandler)
         }
     }
 
-    func checkAuthorization(_ completionHandler: @escaping (Bool) -> Void) {
+    //If a previous attempt to register the device's token failed, this will retry
+    func retryIfTokenNotRegistered() {
+        //If they don't have a wallet yet, don't even try as there is no pub key to register
+        guard TariLib.shared.tariWallet != nil else { return }
+
+        if !hasRegisteredToken {
+            requestAuthorization { (_) in }
+        } else {
+            TariLogger.verbose("Already registered for push notifications")
+        }
+     }
+
+    private func registerWithAPNS(_ completionHandler: @escaping (Bool) -> Void) {
```

Comment:
I like this naming: all caps for initials (some do `Apns`) - it should be the same for Android throughout. And when the abbreviation is at the beginning of a function or var name, something like `apnsToken` - and I guess this one is better in that case rather than `APNSToken`.

Reply:
Ok so this way is fine? Maybe we should start a style guide for the 2 apps, try keep them as close as possible?

Reply:
Yep it's looking good, and style guide definitely!